keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1606 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Add support to Kubernetes for declaring _quality-of-service_ resources, and
+assigning these to Pods. A quality-of-service (QoS-class) resource is similar
+to other Kubernetes resource types (i.e. native resources such as `cpu` and
+`memory` or extended resources) because you can assign that resource to a
+particular container. However, QoS-class resources are also different from
+those other resources because they are used to assign a _class identifier_,
+rather than to declare a specific amount of capacity that is allocated.
+
+Main characteristics of the new resource type (and the technologies they are
+aimed at enabling) are:
+
+- multiple containers can be assigned to the same class of a certain type of
+  QoS-class resource
+- QoS-class resources are represented by an enumerable set of class identifiers
+- each type of QoS-class resource has an independent set of class identifiers
+
+With QoS-class resources, Pods and their containers can request opaque
```

Comment:
The difference between a RuntimeClass and a QoS-class is basically that a RuntimeClass implies a union of common behavior across a variety of dimensions and the goal with QoS-class is to allow a attribute specific mechanism that we trust when unioned together will work well?

Reply:
> QoS-class is to allow a attribute specific mechanism that we trust when unioned together will work well?

Basically yes. It would be assumed that there are no overlapping QoS-class resources for controlling "the same thing". SImilarly that you wouldn't expect two co-exiting device plugins to manage the same devices.